src/realm/object-store/sync/app.cpp

```diff
@@ -819,11 +819,10 @@ void App::update_metadata_and_resend(Request&& request, UniqueFunction<void(cons
          anchor = shared_from_this()](const util::Optional<Response>& response) mutable {
             if (response) {
                 if (util::HTTPStatus(response->http_status_code) == util::HTTPStatus::MovedPermanently) {
```

Comment:
I suggest adding a function like `void App:handle_possible_move_response(Request&& request, const Response& UniqueFunction<void(const Response&)>&& completion)` that encapsultates some of this repeated logic. Then you could just do `if (response) return anchor->handle_possible_move_response(std::move(request), *response, std::move(completion));`

Not necesarily in this review, but I think it would be a nice separate cleanup.